crates/re_space_view_time_series/src/space_view_class.rs

```diff
@@ -170,6 +172,57 @@ impl SpaceViewClass for TimeSeriesSpaceView {
             });
     }
 
+    fn spawn_heuristics(&self, ctx: &ViewerContext<'_>) -> SpaceViewSpawnHeuristics {
+        re_tracing::profile_function!();
+
+        // For all following lookups, checking indicators is enough, since we know that this is enough to infer visualizability here.
+        let Some(indicated_entities) = ctx
+            .indicator_matching_entities_per_visualizer
+            .get(&TimeSeriesSystem::identifier())
+        else {
+            return SpaceViewSpawnHeuristics::default();
+        };
+
+        // Spawn time series data at the root if there's time series data either
+        // directly at the root or one of its children.
+        let subtree_of_root_entity = &ctx.entity_db.tree().children;
+        if indicated_entities.contains(&EntityPath::root())
+            || subtree_of_root_entity
+                .iter()
+                .any(|(_, subtree)| indicated_entities.contains(&subtree.path))
+        {
+            return SpaceViewSpawnHeuristics {
+                recommended_space_views: vec![RecommendedSpaceView {
+                    root: EntityPath::root(),
+                    query_filter: EntityPathFilter::subtree_entity_filter(&EntityPath::root()),
+                }],
+            };
+        }
+
+        // If there's other entities that have the right indicator & didn't match the above,
+        // spawn a time series view for each child of the root that has any entities with the right indicator.
+        let mut child_of_root_entities = HashSet::default();
+        for entity in indicated_entities.iter() {
+            if let Some(child_of_root) = entity.iter().next() {
+                child_of_root_entities.insert(child_of_root);
+            }
+        }
+        let recommended_space_views = child_of_root_entities
+            .into_iter()
+            .map(|path_part| {
+                let entity = EntityPath::new(vec![path_part.clone()]);
+                RecommendedSpaceView {
+                    query_filter: EntityPathFilter::subtree_entity_filter(&entity),
+                    root: entity,
+                }
+            })
+            .collect();
```

Comment:
More weird child-of-root-specific logic makes me sad. I think we should make a choice between:
 - Just use the root (all signals on one chat)
 - Or space-view-per-indicated-entity (each signal on its own chat)

Reply:
Noted this down in
* #4926

and leaving a comment referring to it